### ISSUE TITLE
Slow mail compose with encryption, signature, and large attachment

### ISSUE THREAD

**Issue by Mnkai** [k9mail/k-9#3186]
### Expected behavior
Since modern smartphone's processor is fast, few seconds should be enough to actually encrypt/sign attachment and message itself.

### Actual behavior
After writing message and attaching attachment, K-9 takes about a minute to finish *composing* message and go back to message list with 'sending' notification on actionbar. It finishes sending in acceptable time considering my network condition.

### Steps to reproduce
1. Attach large attachment (> 5MiB) to mail.
2. Select encryption + sign.
3. Press send button.

### Other informations
OpenKeychain file encryption output (identical 5.4MiB JPEG file) with ASCII armor, compression, signature takes 2 seconds.

### Environment
K-9 Mail version: 5.403 (downloaded from Google Play)

Android version: 7.1.2 - LineageOS 14.1 (Oneplus 3T), 8.1 Stock (Nexus 5X)

Account type (IMAP, POP3, WebDAV/Exchange): IMAP/SMTP, but I don't think this is related to this issue.

**Comment by cketti** [comment]
Is this still a problem using the latest versions of K-9 Mail and OpenKeychain?

**Comment by Mnkai** [comment]
Yes, this is still a problem with latest versions of K-9 Mail (5.600) and OpenKeychain (5.2 - build 52009) on Android version 8.0.0 running on Samsung Galaxy A8 2018 (SM-A530).
